pcap4j-core/src/main/java/org/pcap4j/packet/IcmpV6MobilePrefixAdvertisementPacket.java

```diff
@@ -0,0 +1,333 @@
+        private final short reserved;
+        private final List<IpV6NeighborDiscoveryOption> options;
+
+        @SuppressWarnings("squid:S1166")
+        private IcmpV6MobilePrefixAdvertisementHeader(
+                byte[] rawData, int offset, int length) throws IllegalRawDataException {
+            if (length < OPTIONS_OFFSET) {
+                StringBuilder sb = new StringBuilder(120);
+                sb.append("The raw data must be more than ")
+                        .append(OPTIONS_OFFSET - 1).append("bytes")
+                        .append(" to build this header. raw data: ")
+                        .append(ByteArrays.toHexString(rawData, " "))
+                        .append(", offset: ")
+                        .append(offset)
+                        .append(", length: ")
+                        .append(length);
+                throw new IllegalRawDataException(sb.toString());
+            }
+            this.identifier = ByteArrays.getShort(rawData, IDENTIFIER_OFFSET + offset);
+            short tmp = ByteArrays.getShort(rawData, M_O_RESERVED_OFFSET + offset);
+            this.managedAddressConfigurationFlag = (tmp & 0x8000) != 0;
+            this.otherStatefulConfigurationFlag = (tmp & 0x4000) != 0;
+            this.reserved = (short) (0x3FFF & tmp);
+            this.options = new ArrayList<IpV6NeighborDiscoveryOption>();
+            int currentOffsetInHeader = OPTIONS_OFFSET;
+            while (currentOffsetInHeader < length) {
+                IpV6NeighborDiscoveryOptionType type = IpV6NeighborDiscoveryOptionType
+                        .getInstance(rawData[currentOffsetInHeader + offset]);
+                IpV6NeighborDiscoveryOption newOne;
+                try {
+                    newOne = PacketFactories.getFactory(IpV6NeighborDiscoveryOption.class,
+                            IpV6NeighborDiscoveryOptionType.class).newInstance(rawData, currentOffsetInHeader + offset,
+                            length - currentOffsetInHeader, type);
+                } catch (Exception e) {
+                    break;
+                }
+                options.add(newOne);
+                currentOffsetInHeader += newOne.length();
+            }
+        }
+
+        private IcmpV6MobilePrefixAdvertisementHeader(Builder builder) {
+            if ((builder.reserved & 0xC0000000) != 0) {
+                throw new IllegalArgumentException(
+                        "Invalid reserved: " + builder.reserved);
+            }
+            this.identifier = builder.identifier;
+            this.managedAddressConfigurationFlag = builder.managedAddressConfigurationFlag;
+            this.otherStatefulConfigurationFlag = builder.otherStatefulConfigurationFlag;
+            this.reserved = builder.reserved;
+            this.options = new ArrayList<IpV6NeighborDiscoveryOption>(builder.options);
+        }
+
+        public short getIdentifier() {
+            return identifier;
+        }
+
```

Comment:
Please add a method that returns the identifier as int like below:

```java
public int getIdentifierAsInt() {
  return identifier & 0xFFFF;
}
```

Reply:
added.